src/main/java/org/jenkinsci/plugins/workflow/steps/CatchErrorStep.java

```diff
@@ -248,6 +248,7 @@ public Object readResolve() {
                         Functions.printStackTrace(t, listener.getLogger());
                     }
                     if (buildResult.isWorseThan(Result.SUCCESS)) {
+                        listener.getLogger().println("Setting overall build result to " + buildResult);
```

Comment:
shouldn't this only log if the **current** result is worse than this?

e.g. the result is currently `FAILURE`  and buildResult is `UNSTABLE` it would log "setting build result to UNSTABLE" - but this is incorrect as you can not IIRC improve a build result.

e.g. consider the cut down pipeline:

```
stage('outer  stage') {
    catchError(buildResult: 'UNSTABLE') {
      currentBuild.result = "FAILURE"
      error "nope"
    }
}
```

the result of this pipeline is a FAILURE.  and the buildResult should be FAILURE at the time that catchError is intercepted.   (if you remove the `currentBuild.result = "FAILURE"` then this would infact be unstable.)